cliconfig/config.go

```diff
@@ -29,20 +31,15 @@ var (
 	configDir = os.Getenv("DOCKER_CONFIG")
 )
 
-func getDefaultConfigDir(confFile string) string {
-	confDir := filepath.Join(homedir.Get(), confFile)
-	// if the directory doesn't exist, maybe we called docker with sudo
-	if _, err := os.Stat(configDir); err != nil {
-		if os.IsNotExist(err) {
-			return filepath.Join(homedir.GetWithSudoUser(), confFile)
-		}
-	}
-	return confDir
-}
-
 func init() {
 	if configDir == "" {
-		configDir = getDefaultConfigDir(".docker")
+		configDir = filepath.Join(homedir.Get(), defaultConfigDir)
+		if _, err := os.Stat(configDir); err != nil {
+			// if the directory doesn't exist, maybe we called docker with sudo
```

Comment:
ok I guess.  Are there other tools that do this? I'm just curious because if I did:
```
sudo su -
some cool stuff with docker that saved an auth token under .../.docker/
```
I'm not sure we should give that auth token to the non-root user - feels like a security hole.